front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -0,0 +1,194 @@
+    setTimesStopsSteps(suggestedOPs);
+  }, [pathSteps]);
+
+  const columns: Column<PathWaypointColumn>[] = useMemo(
+    () => [
+      {
+        ...keyColumn<PathWaypointColumn, 'name'>('name', createTextColumn<string | undefined>()),
+        title: t('name'),
+        disabled: true,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'ch'>('ch', createTextColumn()),
+        title: 'Ch',
+        disabled: true,
+        grow: 0.1,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'arrival'>('arrival', timeColumn),
+        title: t('arrival_time'),
+        cellClassName: ({ rowIndex }) => (rowIndex === 0 ? 'dsg-hidden-cell' : ''),
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'stop_for'>(
+          'stop_for',
+          createTextColumn({
+            continuousUpdates: false,
+            alignRight: true,
+          })
+        ),
+        title: `${t('stop_time')} (s)`,
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'onStopSignal'>(
+          'onStopSignal',
+          checkboxColumn as Partial<Column<boolean | undefined>>
+        ),
+        title: t('reception_on_closed_signal'),
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'theoreticalMargin'>(
+          'theoreticalMargin',
+          createTextColumn({
+            continuousUpdates: false,
+            alignRight: true,
+            placeholder: t('theoretical_margin_placeholder'),
+            formatBlurredInput: (value) => {
+              if (!value || value === 'none') return '';
+              if (!marginRegExValidation.test(value)) {
+                return `${value}${t('theoretical_margin_placeholder')}`;
+              }
+              return value;
+            },
+          })
+        ),
+        cellClassName: ({ rowData }) => (rowData.isMarginValid ? '' : 'invalidCell'),
+        title: t('theoretical_margin'),
+        disabled: ({ rowIndex }) => rowIndex === findLastIndex(pathProperties.allVias),
+      },
+    ],
+    [t, pathProperties, timesStopsSteps]
+  );
+
+  const removeVia = (
+    rowData: PathWaypointColumn,
+    setRowData: (rowData: PathWaypointColumn) => void
+  ) => {
+    const updatedPathSteps = [...pathSteps];
+    const index = compact(updatedPathSteps).findIndex((step) => {
+      if ('uic' in step) {
+        return step.uic === rowData.uic && step.ch === rowData.ch && step.kp === rowData.kp;
+      }
+      if ('track' in step) {
+        return step.track === rowData.track;
+      }
+      return false;
+    });
+    updatedPathSteps.splice(index, 1);
+
+    setRowData({
+      ...rowData,
+      arrival: undefined,
+      stop_for: undefined,
+      theoreticalMargin: undefined,
+      onStopSignal: false,
+    });
+    dispatch(updatePathSteps(updatedPathSteps));
+  };
+
+  return (
+    <DynamicDataSheetGrid
+      columns={columns as Partial<Column<Partial<PathWaypointColumn>>>[]}
```

Comment:
Do you think you can remove this cast ?
```suggestion
      columns={columns}
```

Reply:
Done